src/unitxt/metrics.py

```diff
@@ -74,6 +74,98 @@ class Metric(Artifact):
     def main_score(self):
         pass
 
+    # Override 'prediction_type' with the expected type of predictions
+    # and references.  Example: "List[str]", "List[Dict]"", "string"
+    prediction_type: string = "Any"
+
+    # Most metrics do not allow None as predictions, but some do (e.g. ndcg)
+    allow_none_prediction: bool = False
+
+    # Standard metrics can receive multiple references per predictions (in a list)
+    # Some metrics support only a single reference per prediction (one element in the list)
+    single_reference_per_prediction: bool = False
+
+    # Some metrics (e.g. HF metrics) do not follow the standard unitxt metric API
+    # where references are passed as list (e.g. in spearmanr references is a scalar).
+    # In this case, and other special cases, we can disable type validation.
+    disable_type_validation: bool = False
```

Comment:
```suggestion
    prediction_type: string
```
The prediction type can detect it all. 
Allow None prediction can be done with:
`prediction_type = Optional[str]`
Single reference should be:
disable_type_validation can be done with:
`prediction_type = Any`
The only one that cannot be done directly is single_reference_per_prediction wchich in my opinion should not be part of this pr.

Reply:
> The prediction type can detect it all. Allow None prediction can be done with: `prediction_type = Optional[str]` Single reference should be: disable_type_validation can be done with: `prediction_type = Any` The only one that cannot be done directly is single_reference_per_prediction wchich in my opinion should not be part of this pr.

Thanks.  I changed the ' prediction type ' to check for None with Optional as you suggested.  Also removed disable_type_validation .   

However 'single_reference_per_prediction' is an important one and exposed some inconsistency in our use of metrics.    unitxt assumes that there are multiple references per predictions.  For example,  in  HF dataset definition (string for predictions, and list of string of references). It then post processes the prediction and each of string references using the post processors.     

In n some metrics (retrieval metrics) the predictions are a list of documents id, and the reference are also a list of document ids .  This means that in few shot learning, only one document will be presented in the example output.  

In multi label , on the other hand, and the predictions are a list of labels and references are a list of list of labels. 
 
Bottom line - all metrics should receive a list of references.  Metrics that can only handle one reference, should set the  'single_reference_per_prediction'  as prediction.
]